### PR TITLE
Adds notes about generating surface dataset and initial condition for ELM

### PR DESCRIPTION
**Pull request by bishtgautam**
Adds notes on using the following ELM tools:

- `mksurfdata_map`: For creating a surface dataset
- `interpinic`: For creating an initial condition.

[BFB]

### CODE REVIEW

File: components/elm/docs/user-guide/interpinic.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#SBATCH  --time 24:00:00
+#SBATCH  -p slurm
+#SBATCH  --account esmd
+
+# Load relevant modules.
+cd <e3sm-dir>
+eval $(./cime/CIME/Tools/get_case_env/get_case_env)
+
+# Change dir to `interpinic`
+cd components/elm/tools/interpinic/src
```
Comment (darincomeau) [inline components/elm/docs/user-guide/interpinic.md:72]: When I tried this, the executable was one directory up (not in /src).

File: components/elm/docs/user-guide/interpinic.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#SBATCH  --job-name=remap
+#SBATCH  --nodes=1
+#SBATCH  --exclusive
+#SBATCH  --time 24:00:00
+#SBATCH  -p slurm
+#SBATCH  --account esmd
+
+# Load relevant modules.
+cd <e3sm-dir>
+eval $(./cime/CIME/Tools/get_case_env/get_case_env)
```
Comment (darincomeau) [inline components/elm/docs/user-guide/interpinic.md:69]: extra 'get_case_env' in path.

File: components/elm/docs/user-guide/interpinic.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+eval $(./cime/CIME/Tools/get_case_env)
+
+# change directory
+cd components/elm/tools/interpinic/src
+
+export USER_LDFLAGS="-L$NETCDF_C_DIR/lib -lnetcdf -L$NETCDF_F_DIR/lib -lnetcdff  -L$HDF5_DIR/lib -lhdf5"
+
+USER_FC=ifort LIB_NETCDF="`nc-config --flibs`" INC_NETCDF="`nf-config --includedir`" make VERBOSE=1
+```
+
```
Comment (chloewhicker) [inline components/elm/docs/user-guide/interpinic.md:51]: cd ..

File: components/elm/docs/user-guide/interpinic.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+# Load relevant modules
+cd <e3sm-dir>
+eval $(./cime/CIME/Tools/get_case_env)
+
+# change directory
+cd components/elm/tools/interpinic/src
+
+export USER_LDFLAGS="-L$NETCDF_C_DIR/lib -lnetcdf -L$NETCDF_F_DIR/lib -lnetcdff  -L$HDF5_DIR/lib -lhdf5"
+
+USER_FC=ifort LIB_NETCDF="`nc-config --flibs`" INC_NETCDF="`nf-config --includedir`" make VERBOSE=1
```
Comment (chloewhicker) [inline components/elm/docs/user-guide/interpinic.md:49]: missing "export" ?

**Comment by rljacob** [comment]
@bishtgautam see comments on this PR.

File: components/elm/docs/user-guide/interpinic.md (hunk 4 above)
Comment (bishtgautam) [reply components/elm/docs/user-guide/interpinic.md:49]: `export` is not required.

File: components/elm/docs/user-guide/interpinic.md (hunk 1 above)
Comment (bishtgautam) [reply components/elm/docs/user-guide/interpinic.md]: fixed.

File: components/elm/docs/user-guide/interpinic.md (hunk 2 above)
Comment (bishtgautam) [reply components/elm/docs/user-guide/interpinic.md]: fixed.

File: components/elm/docs/user-guide/interpinic.md (hunk 3 above)
Comment (bishtgautam) [reply components/elm/docs/user-guide/interpinic.md:53]: fixed.

**Review by peterdschwartz — APPROVED** [review]

**Comment by rljacob** [comment]
@bishtgautam you can merge this.

**Pull request merged**